source/grid/grid_out.cc

```diff
@@ -2951,6 +2951,46 @@ namespace
     return v;
   }
 
+  /**
+   * Return all lines of non-internal faces in three dimension.
+   */
+  std::vector<typename Triangulation<3, 3>::active_line_iterator>
+  all_co_faces(const Triangulation<3, 3> &tria)
+  {
+    std::vector<typename Triangulation<3, 3>::active_line_iterator> res;
+
+    std::vector<bool> flags;
+    tria.save_user_flags_line(flags);
+    const_cast<Triangulation<3, 3> &>(tria).clear_user_flags_line();
+
+    for (auto face = tria.begin_active_face(); face != tria.end_face(); ++face)
+      for (unsigned int l = 0; l < GeometryInfo<3>::lines_per_face; ++l)
+        {
+          auto line = face->line(l);
+          if (line->user_flag_set() || line->has_children())
+            continue;
+          else
+            line->set_user_flag();
+          if (line->boundary_id() != numbers::invalid_boundary_id)
+            res.emplace_back(line);
```

Comment:
doesn't this also catch internal edges?

Reply:
Does it? How should I read than: `line->boundary_id() != numbers::invalid_boundary_id`?

Reply:
I don't recall what `invalid_boundary_id` corresponds to -- are these precisely the interior edges and faces? This might be easier to read if you wrote `if (line->at_boundary())`, assuming that that is what you want.

Reply:
@bangerth done!